pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -1128,17 +1128,34 @@ type TiDBFailureMember struct {
 	CreatedAt metav1.Time `json:"createdAt,omitempty"`
 }
 
+const EvictLeaderAnnKey = "tidb.pingcap.com/evict-leader"
+
+// The `Value` of annotation controls the behavior when the leader count drops to zero, the valid value is one of:
+//
+// - `none`: doing nothing.
+// - `delete-pod`: delete pod and remove the evict-leader scheduler from PD.
+const (
+	EvictLeaderValueNone      = "none"
+	EvictLeaderValueDeletePod = "delete-pod"
+)
+
+type EvictLeaderStatus struct {
+	PodCreateTime metav1.Time `json:"podCreateTime,omitempty"`
+	Value         string      `json:"value,omitempty"`
+}
+
 // TiKVStatus is TiKV status
 type TiKVStatus struct {
-	Synced          bool                        `json:"synced,omitempty"`
-	Phase           MemberPhase                 `json:"phase,omitempty"`
-	BootStrapped    bool                        `json:"bootStrapped,omitempty"`
-	StatefulSet     *apps.StatefulSetStatus     `json:"statefulSet,omitempty"`
-	Stores          map[string]TiKVStore        `json:"stores,omitempty"`
-	PeerStores      map[string]TiKVStore        `json:"peerStores,omitempty"`
-	TombstoneStores map[string]TiKVStore        `json:"tombstoneStores,omitempty"`
-	FailureStores   map[string]TiKVFailureStore `json:"failureStores,omitempty"`
-	Image           string                      `json:"image,omitempty"`
+	Synced          bool                          `json:"synced,omitempty"`
+	Phase           MemberPhase                   `json:"phase,omitempty"`
+	BootStrapped    bool                          `json:"bootStrapped,omitempty"`
+	StatefulSet     *apps.StatefulSetStatus       `json:"statefulSet,omitempty"`
+	Stores          map[string]TiKVStore          `json:"stores,omitempty"`
+	PeerStores      map[string]TiKVStore          `json:"peerStores,omitempty"`
+	TombstoneStores map[string]TiKVStore          `json:"tombstoneStores,omitempty"`
+	FailureStores   map[string]TiKVFailureStore   `json:"failureStores,omitempty"`
+	Image           string                        `json:"image,omitempty"`
+	EvictLeader     map[string]*EvictLeaderStatus `json:"evictLeaderStatus,omitempty"`
```

Comment:
no need to include `Status` in the name as it already defined in `TiKVStatus`?

Reply:
https://github.com/pingcap/tidb-operator/pull/4279/commits/832637aa5fdc8371bdc64c985b0343a3258cfc41